### PR TITLE
Update Deployments

### PR DESCRIPTION
**Pull request by offerijns**

### CODE REVIEW

**Review by offerijns — COMMENTED** [review]
I can't approve as I created the PR, but LGTM!

File: foundry.toml
Comment (NunoAlexandre) [inline foundry.toml:24]: Is this intentional? 🤔 What's the actual consequence of this flag? It seems like a test won't fail if there's an uncaught/unexpected revert

File: src/token/erc20.sol
Comment (NunoAlexandre) [inline src/token/erc20.sol:80]: good catch!

**Review by NunoAlexandre — APPROVED** [review]
Some questions but looks good to me!

File: src/token/factory.sol
Comment (offerijns) [reply src/token/factory.sol:7]: Create2 is an EVM opcode that enables deterministic deployments, see https://docs.openzeppelin.com/cli/2.8/deploying-with-create2

File: src/token/factory.sol
Comment (offerijns) [reply src/token/factory.sol:22]: The inputs to the deterministic address are
- Salt
- Deployer address
- Bytecode
- Arguments

The deployer address here is key. If we deploy the factory address, then any tranche tokens deployed from that factory address will be deterministic. If someone else deploys a tranche token with the same bytecode, salt and arguments, but from their own address or factory, then the deployer address will not match and it will lead to different addresses.

File: test/token/factory.sol
Comment (offerijns) [reply test/token/factory.sol:47]: Yeah this test won't work until deployments are actually deterministic, so Adam will work on this in a PR after merging this.

File: src/token/factory.sol
Comment (NunoAlexandre) [reply src/token/factory.sol:22]: Nice 👌

**Pull request merged**
